dfsclient/src/test/java/org/hpccsystems/dfs/client/DFSReadWriteTest.java

```diff
@@ -451,6 +448,59 @@ public void numericOverflowTest() throws Exception
         }
     }
 
+    final static String dimdatefilename = "dfsclient::junit::dim_date";
+    @Test
+    public void filteredDIMDATEJAPI445Test() throws Exception
+    {
+        List<HPCCRecord> records = new ArrayList<HPCCRecord>();
+        FieldDef[] fieldDefs = new FieldDef[2];
+        fieldDefs[0] = new FieldDef("date_sk", FieldType.INTEGER, "unsigned8", 8, true, false, HpccSrcType.LITTLE_ENDIAN, new FieldDef[0]);
+        fieldDefs[1] = new FieldDef("day_of_wk_num", FieldType.INTEGER, "unsigned8", 8, false, false, HpccSrcType.UTF8, new FieldDef[0]);
+        FieldDef recordDef = new FieldDef("RootRecord", FieldType.RECORD, "rec", 4, false, false, HpccSrcType.LITTLE_ENDIAN, fieldDefs);
+
+        for (int i = 0; i < 10; i++)
+        {
+            Calendar cal = Calendar.getInstance();
+            cal.set(Calendar.YEAR, 2021);
+            cal.set(Calendar.DAY_OF_YEAR, RANDOM.nextInt(365));
+
+            Object[] fields = new Object[2];
+            fields[0] = cal.get(Calendar.YEAR)*10000 + cal.get(Calendar.MONTH)*100 + cal.get(Calendar.DAY_OF_MONTH);
+            fields[1] = cal.get(Calendar.DAY_OF_YEAR);
+            HPCCRecord record = new HPCCRecord(fields, recordDef);
+            records.add(record);
+        }
+        writeFile(records, dimdatefilename, recordDef, connTO);
+
+        HPCCFile file = new HPCCFile(dimdatefilename, connString , hpccUser, hpccPass);
+        file.setFilter("date_sk >= 20200701 and date_sk <= 20211231");
+
+        records = readFile(file, connTO, false);
+        if (records.size() == 0)
```

Comment:
It might be a good idea to check if it can return a specific set of records as well.